### PR TITLE
Security analysis functional logs

### PR DESCRIPTION
**Pull request by geofjamg**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
There is no functional logs for security analysis

**What is the new behavior (if this is a feature change)?**
Minimal functionnal logs have been added 

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/test/resources/saReport.txt [hunk 1]
```diff
@@ -0,0 +1,31 @@
++ Test security analysis report
+  + AC security analysis on network 'sim1'
+    + Network CC0 SC0
+      + Post loading processing
+         Network has 4 buses and 4 branches
+         Network balance: active generation=607.0 MW, active load=600.0 MW, reactive generation=0.0 MVar, reactive load=200.0 MVar
+      + Pre-contingency simulation
+         AC load flow complete with NR status 'CONVERGED'
```
Comment (geofjamg) [inline src/test/resources/saReport.txt:8]: @annetill @flo-dup here we can see an issue with reporter API, we don't keep sequence of reports and sub reporters

File: src/test/resources/saReport.txt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        + Outer loop Distributed slack on generation
+           Iteration 0: failed to distribute slack bus active power mismatch, -1.4404045651219555 MW remains
+        + Outer loop Reactive limits
```
Comment (geofjamg) [inline src/test/resources/saReport.txt:11]: Maybe here another possible improvement: should we discard empty child reporters?

**Comment by geofjamg** [comment]
https://github.com/powsybl/powsybl-core/issues/2190
https://github.com/powsybl/powsybl-core/issues/2190
https://github.com/powsybl/powsybl-core/issues/2190

File: src/main/java/com/powsybl/openloadflow/network/LfNetwork.java [hunk 3]
```diff
@@ -107,6 +107,10 @@ public Reporter getReporter() {
         return reporter;
     }
 
+    public void setReporter(Reporter reporter) {
```
Comment (flo-dup) [inline src/main/java/com/powsybl/openloadflow/network/LfNetwork.java:110]: Here you're breaking in a way the way the reporter was at first designed... It sure is handy to put it in LfNetwork but it was first thought to be passed along all the methods and not retained, in the javadoc:

> A reporter is not meant to be shared with other threads nor to be saved as a class parameter

Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/network/LfNetwork.java:110]: A LF network won't ever be shared between multiple thread as it is used as working data (including writing working data)

**Review by flo-dup — APPROVED** [review]

**Pull request merged**